src/Core_Fix_SceneLoad/PageKeeper.cs

```diff
@@ -0,0 +1,85 @@
+﻿#if KK || KKS
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using HarmonyLib;
+using UnityEngine;
+using Studio;
+using UnityEngine.UI;
+
+
+namespace IllusionFixes
+{
+    /// <summary>
+    /// Fixes a problem in the UI for loading scenes where page and scrollbar positions are not saved.
+    /// This seems to occur only on KK/KKS.
+    /// </summary>
+    public class PageKeeper : MonoBehaviour
+    {
+        static float _scrollValue = 1f;
+        static int _page = 0;
+
+        Scrollbar _target;
+
+#if KK
+        int _startFrame;
+        SceneLoadScene _parent;
+#endif
+
+        public static void Setup()
+        {
+            Harmony.CreateAndPatchAll(typeof(PageKeeper), FixSceneLoad.GUID + " PageKeeper");
+        }
+
+        [HarmonyPostfix]
+        [HarmonyPatch(typeof(SceneLoadScene), "Awake")]
+        internal static void PostSceneLoadSceneAwake(SceneLoadScene __instance)
+        {
+            var gboj = __instance?.gameObject;
+            if (gboj != null)
+                gboj.AddComponent<PageKeeper>();
+        }
+
+        void Start()
+        {
+            _target = gameObject.GetComponentInChildren<Scrollbar>();
+
+            if (_target == null)
+            {
+                Destroy(this);
+                return;
+            }
+
+            _target.value = _scrollValue;
+
+#if KK
+            _startFrame = Time.frameCount;
+            _parent = GetComponent<SceneLoadScene>();
+#endif
+        }
+
+#if KK
+        void LateUpdate()
+        {
+            //The page is reset by someone else, so set

```

Comment:
This is caused by the [StudioSceneNavigation](https://github.com/GeBo1/GeBoPlugins/blob/master/src/StudioSceneNavigation/README.md) plugin. It switches the page to the page of last loaded scene file.

I think it's better to not set the page in this plugin, and instead let the user decide if they want to keep StudioSceneNavigation or remove it.

Reply:
Ahhhh, so this plugin is reverting to the page with the last loaded scene, not the first page.
I agree to turn off the page function.

Reply:
Also, keeping the page is a very minor optional feature of StudioSceneNavigation, so they can just turn off `Restore Loader Page` in options